src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectISizeDistributionSpark.java

```diff
@@ -0,0 +1,175 @@
+    public boolean properlyPairedOnly = true;
+
+    @Argument(shortName = "U",
+            fullName = "mappedReadsOnly",
+            doc = "If set to true, include mapped reads only. Default value: true.")
+    public boolean mappedReadsOnly = true;
+
+    @Argument(shortName = "Um",
+            fullName = "mateMappedReadsOnly",
+            doc = "If set to true, include mate-mapped reads only. Default value: true.")
+    public boolean mateMappedReadsOnly = true;
+
+    @Argument(shortName = "D",
+            fullName = "nonDupReadsOnly",
+            doc = "If set to true, include non-duplicated reads only. Default value: true.")
+    public boolean nonDupReadsOnly = true;
+
+    @Argument(shortName = "S",
+            fullName = "nonSecondaryAlignmentsOnly",
+            doc = "If set to true, filter out secondary alignments reads. Default value: true.")
+    public boolean nonSecondaryAlignmentsOnly = true;
+
+    @Argument(shortName = "SS",
+            fullName = "nonSupplementaryAlignmentsOnly",
+            doc = "If set to true, filter out supplementary alignments reads. Default value: true.")
+    public boolean nonSupplementaryAlignmentsOnly = true;
+
+    @Argument(shortName = "F",
+            fullName = "whichEndOfPairToUse",
+            doc = "Which end of pairs to use for collecting information. Possible values:{0,1,2}, where 1 and 2 " +
+                    "stands for first or second end respectively. Option 0 picks up information from 1st end " +
+                    "when both ends are available, and pick either one when only one end is available. " +
+                    "(Remember, in SV analysis, a truncated region may be investigated, so this is possible.)" +
+                    " Unused if only paired reads are used (properlyPairedOnly==true).")
+    public int whichEndOfPairToUse = 1;
+
+    @Argument(shortName = "Q",
+            fullName = "MQPassingThreshold",
+            doc = "If set non-zero value, only include reads passing certain mapping quality threshold.")
+    public double MQPassingThreshold = 0.0;
+
+    @Argument(shortName = "Qm",
+            fullName = "mateMQPassingThreshold",
+            doc = "If set to non-zero value, only include reads with mate passing certain mapping quality threshold.")
+    public double mateMQPassingThreshold = 0.0; // TODO: efficient way to test
+
+
+    // path to Picard R script for producing histograms in PDF files.
+    private static final String R_SCRIPT = "insertSizeHistogram.R";
+
+    // should this be static?
+    // Calculates InsertSizeMetrics for all METRIC_ACCUMULATION_LEVELs provided
+    private static ISZMetricsCollectorSpark collector;
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final JavaRDD<GATKRead> reads = getReads();
+
+        collector = new ISZMetricsCollectorSpark(reads, METRIC_ACCUMULATION_LEVEL, getHeaderForReads().getReadGroups(), DEVIATIONS_TOL);
+
+        if(PRODUCE_FILES)
```

Comment:
always use braces for control flow statements like `if`,`for`,`while`, etc. The only exception are one-line ifs + return or if + throw but they must be on 1 line - the if and the rest and